src/store/app/thunks.test.ts

```diff
@@ -87,13 +135,145 @@ describe("thunks", () => {
+      rootStateFactory.build({
+        general: generalStateFactory.build({
+          config: configFactory.build({
+            isJuju: true,
+          }),
+          controllerConnections: {
+            "wss://example.com": {
+              user: {
+                "display-name": "eggman",
+                identity: "user-eggman@external",
+                "controller-access": "",
+                "model-access": "",
+              },
+            },
+          },
+          credentials: {
+            "wss://example.com": credentialFactory.build(),
+          },
+        }),
+        juju: jujuStateFactory.build({
+          controllers: {
+            "wss://example.com": [
+              controllerFactory.build({
+                path: "/",
+                uuid: "uuid123",
+                version: "1",
+              }),
+            ],
+          },
+        }),
+      }),
+    );
+    const action = connectAndStartPolling();
+    await action(dispatch, getState, null);
+    expect(dispatch).toHaveBeenCalledWith(
+      appActions.connectAndPollControllers({
+        controllers: [["wss://controller.example.com", undefined, false]],
+        isJuju: true,
+      }),
+    );
+    expect(console.error).toHaveBeenCalledWith(
+      "Error while triggering the connection and polling of models.",
+      new Error("Error while dispatching connectAndPollControllers!"),
+    );
+    expect(dispatch).toHaveBeenCalledWith(
+      generalActions.storeConnectionError(
+        "Unable to connect: Error while dispatching connectAndPollControllers!",
+      ),
+    );
+  });
+
+  it("connectAndStartPolling should catch non-standard type of error", async () => {
+    const dispatch = jest
+      .fn()
+      // Successfuly dispatch connectAndStartPolling/pending.
+      .mockImplementationOnce(() => {})
+      // Throw error when trying to dispatch connectAndPollControllers.
+      .mockImplementationOnce(() => {
+        // eslint-disable-next-line no-throw-literal
+        throw ["Unknown error."];
+      });
+    const getState = jest.fn(() =>
+      rootStateFactory.build({
+        general: generalStateFactory.build({
+          config: configFactory.build({
+            isJuju: true,
+          }),
+          controllerConnections: {
+            "wss://example.com": {
+              user: {
+                "display-name": "eggman",
+                identity: "user-eggman@external",
+                "controller-access": "",
+                "model-access": "",
+              },
+            },
+          },
+          credentials: {
+            "wss://example.com": credentialFactory.build(),
+          },
+        }),
+        juju: jujuStateFactory.build({
+          controllers: {
+            "wss://example.com": [
+              controllerFactory.build({
+                path: "/",
+                uuid: "uuid123",
+                version: "1",
+              }),
+            ],
+          },
+        }),
+      }),
```

Comment:
It looks like this same state is defined a number of times in this file. Maybe it could be defined once in the `beforeEach`.

Reply:
Done! Nice catch :)